packages/maskbook/src/plugins/BarnBridge/Model/SYPoolModel.ts

```diff
@@ -0,0 +1,81 @@
+import { BB_SY_CREAM, BB_SY_AAVE, BB_SY_COMPOUND, APP_URL, SY_URL_FRAGMENT } from '../constants'
+import type { SYCoinProps } from '../UI/SmartYieldPoolView'
+import { useAsyncRetry } from 'react-use'
+import urlcat from 'urlcat'
+
+export type SYPoolModelData = { [id: string]: SYCoinProps[] }
+
+export function SmartYieldPoolModelGetData() {
+    return useAsyncRetry(async () => {
+        const response = await fetch('https://api-v2.barnbridge.com/api/smartyield/pools?originator=all', {
```

Comment:
```suggestion
        const response = await fetch(urlcat(API_URL, '/api/smartyield/pools', { originator: 'all' }), {
```

Reply:
Done! @septs  if you can figure out why in the world my transition to Worker module isn't working I'll love you forever